clients/wallet/src/stellar_wallet.rs

```diff
@@ -41,6 +47,10 @@ pub struct StellarWallet {
 
 	/// the waiting time (in seconds) for retrying.
 	max_backoff_delay: u16,
+
+	/// a client to connect to Horizon
+	// todo: it's probably better that this is of generic type C, to remove dependency from reqwest
```

Comment:
Can we do what the 'todo' says already? Or is there any blocker?

Reply:
:sob: it's a bit difficult, what with the wallet being cloned and shared in different tokio spawns; an example would be the code in `fn resubmit_transactions_from_cache()` in stellar_wallet.rs.

Thinking about it, I guess it doesn't matter right now, since the client is private anyway. And it is instantiated inside this crate, so external code won't even need to bother. 

I'll just remove the `todo`.